### DISCUSSION TITLE
Using chart diagrams inside the Pager for composes consume swipe gestures

### DISCUSSION THREAD

**Discussion by EvgenySamarin** [patrykandpatrick/vico#765] [Ideas]
You have a great library 👍🏼, but I ran into a problem when using multiple charts inside the Horizontal pager. 

Since you are using `pointerInput` and `detectHorizontalDragGesture` inside `chartTouchEvent`, the parent component cannot intercept the swipe event, which is why the pager stands still. 

Could you replace `detectHorizontalDragGesture` with `detectDragGestureAfterLongPress`. This will allow users to view markers on the diagram, and it also allow embedding diagrams in any swipable containers, which will add flexibility to your library

`pointerInteropFilter` also solves the problem, but using it makes the interface less responsive. 

<img width="755" alt="image" src="https://github.com/patrykandpatrick/vico/assets/36430830/76b3b39f-9559-47d6-8d1c-acbd3f68333b">

**Comment by Gowsky** [comment]
Hi @EvgenySamarin, [Vico 2.0.0-alpha.22](https://patrykandpatrick.com/vico/releases/2.0.0-alpha.22) fixes the nested scroll issue, thus charts can be used inside of horizontal pagers without any gesture issues. We'll look into making markers appear on long press. Also, please see #355.
